atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/pue/ResilientCommitTimestampPutUnlessExistsTable.java

```diff
@@ -28,18 +31,36 @@
 import com.palantir.common.streams.KeyedStream;
 import com.palantir.logsafe.Preconditions;
 import com.palantir.logsafe.SafeArg;
-import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
+import javax.annotation.Nonnull;
+import org.immutables.value.Value;
 
 public class ResilientCommitTimestampPutUnlessExistsTable implements PutUnlessExistsTable<Long, Long> {
+    private static final int TOUCH_CACHE_SIZE = 1000;
+
     private final ConsensusForgettingStore store;
     private final TwoPhaseEncodingStrategy encodingStrategy;
     private final Supplier<Boolean> acceptStagingReadsAsCommitted;
 
+    private final LoadingCache<CellInfo, Long> touchCache = CacheBuilder.newBuilder()
```

Comment:
Can we use Caffeine instead of Guava caches? Baseline checkstyle should flag this https://github.com/palantir/gradle-baseline/blob/ca93cc1fd072ebcf00866364ba85f8f0d1d63d20/.baseline/checkstyle/checkstyle.xml

Reply:
I knew one was preferred, but wasn't sure which one and this was used elsewhere. Thanks for flagging 👍